### ISSUE TITLE
Unclear what client is used for coordination primitives

### ISSUE THREAD

**Issue by fuglede** [dask/distributed#3343]
I'm currently working with the coordination primitives in `dask.distributed`. First, let us note that the following code blows up:

```
from dask.distributed import Queue
q = Queue()
...
AttributeError: 'NoneType' object has no attribute 'asynchronous'
```
This makes sense, since clearly we need something that can talk to the scheduler. The docs provide the following example:
```
from dask.distributed import Client, Queue
c = Client()
q = Queue()
```
With this, `q.client` is `c` which also makes sense, because what else could it be. Now, consider the following example:
```
from dask.distributed import Client, Queue
c1 = Client()
c2 = Client()
q = Queue()
```
Here, there are two candidates for what `q.client` could be. It ends up being `c2` but from the documentation, it's not clear what sort of logic is applied: Does Dask keep track of all `Client`s created in the current process, then picks the last one? In any case, it would be nice with just a couple words explaining this in the docs.

**Comment by mrocklin** [comment]
> Does Dask keep track of all Clients created in the current process, then picks the last one?

Yes.  See the `default_client` method.

> it would be nice with just a couple words explaining this in the docs.

Where in the docs would you expect to see this?

Note also that if you want to be explicit you can provide a `client=` keyword to the `Queue` constructor

**Comment by fuglede** [comment]
> Yes. See the `default_client` method.

Gotcha, thanks!

> Where in the docs would you expect to see this?

Good question. The first place I looked was in the docstring for `Queue` and its `client` parameter, so that's what I would have found most natural. Then, of course, you would have to add it for all the other primitives as well though. The second place I looked was in the [Coordinate Primitives section of the futures sections](https://docs.dask.org/en/latest/futures.html#coordination-primitives), and I think just having it there would be fine as well. If you agree that this would be useful, and if you don't already have a phrasing in mind, I would be happy to provide a PR.
